CRM/SMS/BAO/Provider.php

```diff
@@ -87,15 +88,20 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   }
 
   /**
+   * Save a new record into the database
+   * @todo create a create function to do this work
    * @param $values
    */
   public static function saveRecord($values) {
+    $values['domain_id'] = CRM_Utils_Array::value('domain_id', $values, CRM_Core_Config::domainID());
```

Comment:
we discussed that with the api the default can be set & overridden at the api level rather than here